Patch methods do not correctly resolve generic collection names withEntities

### Which @ngrx/* package(s) are the source of the bug?

signals

### Minimal reproduction of the bug/regression with instructions

 A TypeScript error occurs when attempting to use the patchState function with addEntity on a store with `withEntities`. The error indicates a type mismatch between PartialStateUpdater\<NamedEntityState\<E, Collection\>\> and the expected parameter type.

```ts
import {
  patchState,
  signalStoreFeature,
  type,
  withMethods,
} from '@ngrx/signals';
import { addEntity, EntityId, withEntities } from '@ngrx/signals/entities';

function withFeature<E extends { id: EntityId }, Collection extends string>(
  collection: Collection
  // collection: string,  //this removes the errors
) {
  const config = {
    collection: collection,
    entity: type<E>(),
  };
  return signalStoreFeature(
    withEntities(config),
    withMethods((store) => ({
      add: (entity: E) =>
        patchState(
          store,
          addEntity(entity, { collection })
        ),
    }))
  );
}
```


```
X [ERROR] TS2345: Argument of type 'PartialStateUpdater<NamedEntityState<E, Collection>>' is not assignable to parameter of type 'Partial<{ [
K in keyof { [K in keyof NamedEntityState<E, Collection>]: NamedEntityState<E, Collection>[K]; }]: { [K in keyof NamedEntityState<E, Collection>]: NamedEntityState<...>[K]; }[K]; }> | PartialStateUpdater<...>'.
  Type 'PartialStateUpdater<NamedEntityState<E, Collection>>' is not assignable to type 'PartialStateUpdater<{ [K in keyof { [K in keyof Name
dEntityState<E, Collection>]: NamedEntityState<E, Collection>[K]; }]: { [K in keyof NamedEntityState<E, Collection>]: NamedEntityState<...>[K]; }[K]; }>'.
    Types of parameters 'state' and 'state' are incompatible.
      Type '{ [K in keyof { [K in keyof NamedEntityState<E, Collection>]: NamedEntityState<E, Collection>[K]; }]: { [K in keyof NamedEntityState<E, Collection>]: NamedEntityState<...>[K]; }[K]; }' is not assignable to type 'NamedEntityState<E, Collection>'.
        Type 'string' is not assignable to type 'keyof NamedEntityState<E, Collection>'.
          Type 'string' is not assignable to type 'keyof NamedEntityState<E, Collection>'.
            Type '"entityMap"' is not assignable to type '`${Collection}${Capitalize<K>}` | `${Collection}EntityMap` | `${Collection}Ids`'. [plugin angular-compiler]
```

https://stackblitz.com/edit/vitejs-vite-nuijcczs?file=src%2Fmain.ts

### Expected behavior

`Collection extends string` generic could be used to make this feature type-safe and allow method naming like in 
[ngrx-toolkit](https://github.com/angular-architects/ngrx-toolkit) withDataService

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

"@ngrx/signals": "^19.0.0",


### Other information

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

--- a/modules/signals/entities/spec/types/with-entities.types.spec.ts
+++ b/modules/signals/entities/spec/types/with-entities.types.spec.ts
@@ -1,0 +1,51 @@
+import { expecter } from 'ts-snippet';
+import { compilerOptions } from './helpers';
+
+describe('withEntities', () => {
+  const expectSnippet = expecter(
+    (code) => `
+        import {
+          patchState,
+          signalStoreFeature,
+          type,
+          withMethods,
+        } from '@ngrx/signals';
+        import {
+          addEntity,
+          entityConfig,
+          EntityId,
+          withEntities,
+        } from '@ngrx/signals/entities';
+
+        ${code}
+      `,
+    compilerOptions()
+  );
+
+  it('succeeds when creating a custom feature with named collection', () => {
+    const snippet = `
+      function withAddEntities<
+        Entity extends { id: EntityId },
+        Collection extends string
+      >(
+        collection: Collection
+      ) {
+        const config = entityConfig({
+          entity: type<Entity>(),
+          collection,
+        });
+
+        return signalStoreFeature(
+          withEntities(config),
+          withMethods((store) => ({
+            addEntity(entity: Entity): void {
+              patchState(store, addEntity(entity, { collection }));
+            },
+          }))
+        );
+      }
+    `;
+
+    expectSnippet(snippet).toSucceed();
+  });
+});

--- a/modules/signals/spec/types/patch-state.types.spec.ts
+++ b/modules/signals/spec/types/patch-state.types.spec.ts
@@ -36,19 +36,19 @@ describe('patchState', () => {
 
   it('fails with wrong partial state object', () => {
     expectSnippet('patchState(state, { x: 1 })').toFail(
-      /'x' does not exist in type 'Partial<{ count: number; foo: string; }>/
+      /'x' does not exist in type 'Partial<NoInfer<{ count: number; foo: string; }>>/
     );
     expectSnippet("patchState(state, { foo: 'baz' }, { x: 1 })").toFail(
-      /'x' does not exist in type 'Partial<{ count: number; foo: string; }>/
+      /'x' does not exist in type 'Partial<NoInfer<{ count: number; foo: string; }>>/
     );
     expectSnippet('patchState(state, { x: 1 }, { count: 0 })').toFail(
-      /'x' does not exist in type 'Partial<{ count: number; foo: string; }>/
+      /'x' does not exist in type 'Partial<NoInfer<{ count: number; foo: string; }>>/
     );
     expectSnippet('patchState(state, increment(), { x: 1 })').toFail(
-      /'x' does not exist in type 'Partial<{ count: number; foo: string; }>/
+      /'x' does not exist in type 'Partial<NoInfer<{ count: number; foo: string; }>>/
     );
     expectSnippet('patchState(state, { x: 1 }, increment())').toFail(
-      /'x' does not exist in type 'Partial<{ count: number; foo: string; }>/
+      /'x' does not exist in type 'Partial<NoInfer<{ count: number; foo: string; }>>/
     );
   });
 

--- a/modules/signals/src/state-source.ts
+++ b/modules/signals/src/state-source.ts
@@ -8,7 +8,6 @@ import {
   untracked,
   WritableSignal,
 } from '@angular/core';
-import { Prettify } from './ts-helpers';
 
 declare const ngDevMode: unknown;
 
@@ -64,8 +63,7 @@ export function isWritableStateSource<State extends object>(
 export function patchState<State extends object>(
   stateSource: WritableStateSource<State>,
   ...updaters: Array<
-    | Partial<Prettify<NoInfer<State>>>
-    | PartialStateUpdater<Prettify<NoInfer<State>>>
+    Partial<NoInfer<State>> | PartialStateUpdater<NoInfer<State>>
   >
 ): void {
   const currentState = untracked(() => getState(stateSource));

--- a/modules/signals/src/with-feature.ts
+++ b/modules/signals/src/with-feature.ts
@@ -37,7 +37,7 @@ export function withFeature<
       StateSignals<Input['state']> &
         Input['props'] &
         Input['methods'] &
-        WritableStateSource<Prettify<Input['state']>>
+        WritableStateSource<Input['state']>
     >
   ) => SignalStoreFeature<Input, Output>
 ): SignalStoreFeature<Input, Output> {

--- a/modules/signals/src/with-hooks.ts
+++ b/modules/signals/src/with-hooks.ts
@@ -12,7 +12,7 @@ type HookFn<Input extends SignalStoreFeatureResult> = (
     StateSignals<Input['state']> &
       Input['props'] &
       Input['methods'] &
-      WritableStateSource<Prettify<Input['state']>>
+      WritableStateSource<Input['state']>
   >
 ) => void;
 
@@ -21,7 +21,7 @@ type HooksFactory<Input extends SignalStoreFeatureResult> = (
     StateSignals<Input['state']> &
       Input['props'] &
       Input['methods'] &
-      WritableStateSource<Prettify<Input['state']>>
+      WritableStateSource<Input['state']>
   >
 ) => {
   onInit?: () => void;

--- a/modules/signals/src/with-methods.ts
+++ b/modules/signals/src/with-methods.ts
@@ -19,7 +19,7 @@ export function withMethods<
       StateSignals<Input['state']> &
         Input['props'] &
         Input['methods'] &
-        WritableStateSource<Prettify<Input['state']>>
+        WritableStateSource<Input['state']>
     >
   ) => Methods
 ): SignalStoreFeature<Input, { state: {}; props: {}; methods: Methods }> {

--- a/modules/signals/src/with-props.ts
+++ b/modules/signals/src/with-props.ts
@@ -17,7 +17,7 @@ export function withProps<
       StateSignals<Input['state']> &
         Input['props'] &
         Input['methods'] &
-        WritableStateSource<Prettify<Input['state']>>
+        WritableStateSource<Input['state']>
     >
   ) => Props
 ): SignalStoreFeature<Input, { state: {}; props: Props; methods: {} }> {

